packages/databyss-services/sources/SourcesProvider.js

```diff
@@ -0,0 +1,72 @@
+import React, { createContext, useContext, useState, useEffect } from 'react'
+import { Text, View } from '@databyss-org/ui/primitives'
+import Loading from '@databyss-org/ui/components/Loading'
+import createReducer from '@databyss-org/services/lib/createReducer'
+import { SourceNotFoundError } from './SourceNotFoundError'
+import { initialState } from './reducer'
+import { fetchSource, saveSource } from './actions'
+
+const useReducer = createReducer()
+
+export const SourceContext = createContext()
+
+const Error = () => (
```

Comment:
Let's move this to `databyss-ui/components/Notify/ErrorFallback` and make it take a `message` prop